tests/unit/package_managers/test_npm.py

```diff
@@ -428,20 +429,112 @@ def test_get_packages(
         assert package_lock.workspaces == expected_workspaces
         assert package_lock.main_package == expected_main_package
 
-    def test_get_sbom_components(self) -> None:
+    @pytest.mark.parametrize(
+        "packages, expected_components",
+        [
+            (
+                [
+                    Package(
+                        "foo",
+                        "/path/to/node_modules/foo",
+                        {
+                            "version": "1.0.0",
+                            "resolved": "https://github.com/foo",
+                        },
+                    ),
+                ],
+                [
+                    {
+                        "name": "foo",
+                        "missing_hash_in_file": Path("path/to/node_modules/foo/package-lock.json"),
+                    },
```

Comment:
I still find these tests confusing. `/path/to/node_modules/foo` is the path to the installed `foo` dependency. Why is missing_hash_in_file = `path/to/node_modules/foo/package-lock.json`?

Reply:
yup, that's nonsense, should I keep the changes? I think we don't need this anymore since it's tested in `test_resolve_npm` .

Reply:
Does test_resolve_npm cover git dependencies (we don't want to report missing_hash) and https dependencies (we do)? If yes, that's enough IMO

Reply:
there was some missing, added a test case